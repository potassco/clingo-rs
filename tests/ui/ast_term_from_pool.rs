// ast_term_from_pool
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let args = vec![term1, term2];
    let mut pool = ast::pool(&loc, &args).unwrap();
    let term3: ast::Term = pool.into();

    pool = ast::pool(&loc, &args).unwrap();
    drop(args);
    let _end = (pool, term3);
}
