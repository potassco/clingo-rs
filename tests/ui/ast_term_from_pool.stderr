error[E0505]: cannot move out of `args` because it is borrowed
  --> tests/ui/ast_term_from_pool.rs:14:10
   |
9  |     let args = vec![term1, term2];
   |         ---- binding `args` declared here
10 |     let mut pool = ast::pool(&loc, &args).unwrap();
   |                                    ----- borrow of `args` occurs here
...
14 |     drop(args);
   |          ^^^^ move out of `args` occurs here
15 |     let _end = (pool, term3);
   |                 ---- borrow later used here
